yktrace/src/hwt/mapper.rs

```diff
@@ -2,73 +2,66 @@ use object::Object;
 use phdrs::objects;
 
 use crate::SirLoc;
-use hwtracer::Trace;
+use hwtracer::{HWTracerError, Trace};
+use lazy_static::lazy_static;
 use std::{borrow, env, fs};
 
-pub struct HWTMapper {
-    phdr_offset: u64,
+lazy_static! {
     /// Maps a label address to its symbol name and block index.
     ///
     /// We use a vector here since we never actually look up entries by address; we only iterate
     /// over the labels checking if each address is within the range of a block.
-    labels: Option<Vec<(u64, (String, u32))>>
+    ///
+    /// The labels are the same for each trace, and they are immutable, so it makes sense for this
+    /// to be a lazy static, loaded only once and shared.
```

Comment:
What about `dlopen`? Or do you plan to simply not support it?

Reply:
Not for now, no. We will have to eventually come back to it though.

Reply:
Maybe add a fixme?

Reply:
Fixed in 1c5e833